common/src/main/java/com/google/udmi/util/SiteModel.java

```diff
@@ -275,12 +277,13 @@ public static List<String> listDevices(File devicesDir) {
       return ImmutableList.of();
     }
     String[] devices = requireNonNull(devicesDir.list());
-    return Arrays.stream(devices).filter(SiteModel::validDeviceDirectory)
+    return Arrays.stream(devices).filter(device -> validDeviceDirectory(devicesDir, device))
         .collect(Collectors.toList());
   }
 
-  private static boolean validDeviceDirectory(String dirName) {
-    return !(dirName.startsWith(".") || dirName.endsWith("~"));
+  private static boolean validDeviceDirectory(File baseDir, String dirName) {
+    return !(dirName.startsWith(".") || dirName.endsWith("~")) &&
```

Comment:
The bug is correct in asserting this should be a "warning" of some sort, but I'm not sure we currently have a mechanism in place to have site-model level warnings (or maybe we do, I honestly can't remember). But, if not, I think a better intermediate solution would be to just have a more clear error message?

I can't think of any legit reason to have a stray file in that directory.

I actually can't even remember why this logic is here in the first place -- e.g. the `startsWith` and `endsWith` checks -- seems like those should be errors/warnings as well..